pypesto/visualize/observable_mapping.py

```diff
@@ -37,10 +41,346 @@
+            pypesto_result.optimize_result.list[start_index]["x"],
+        )
+    )
+
+    x_dct.update(
+        pypesto_result.problem.objective.calculator.necessary_par_dummy_values
+    )
+
+    # Fill in the parameters.
+    fill_in_parameters(
+        edatas=edatas,
+        problem_parameters=x_dct,
+        scaled_parameters=True,
+        parameter_mapping=parameter_mapping,
+        amici_model=amici_model,
+    )
+
+    # Simulate the model with the parameters from the pypesto result.
+    inner_rdatas = amici.runAmiciSimulations(
+        amici_model,
+        amici_solver,
+        edatas,
+        num_threads=min(n_threads, len(edatas)),
+    )
+
+    # If any amici simulation failed, raise warning and return None.
+    if any(rdata.status != amici.AMICI_SUCCESS for rdata in inner_rdatas):
+        warnings.warn(
+            "Warning: Some AMICI simulations failed. Cannot plot inner "
+            "solutions.",
+            stacklevel=2,
+        )
+        return None
+
+    # Get the simulation.
+    sim = [rdata[AMICI_Y] for rdata in inner_rdatas]
+
+    # Get the inner parameters from the pypesto result.
+    inner_parameter_values = dict(
+        zip(
+            pypesto_problem.inner_x_names,
+            pypesto_result.optimize_result.list[start_index][INNER_PARAMETERS],
+        )
+    )
+
+    ######################################
+    # Plot the linear observable mappings.
+    ######################################
+
+    # for each group, plot the linear observable mapping
+    for observable_index, observable_id in zip(
+        relative_observable_indices, relative_observable_ids
+    ):
+        # Get the ax for the current observable.
+        if rel_and_semiquant_obs_indices is not None:
+            ax_index = rel_and_semiquant_obs_indices.index(observable_index)
+        else:
+            ax_index = relative_observable_indices.index(observable_index)
+
+        ax = axes[ax_index]
+
+        # Get the inner parameters for the current observable.
+        inner_parameters = inner_problem.get_xs_for_group(observable_index + 1)
+
+        scaling_factor = None
+        offset = None
+
+        for inner_par in inner_parameters:
+            if inner_par.inner_parameter_type == InnerParameterType.SCALING:
+                scaling_factor = inner_par
+            elif inner_par.inner_parameter_type == InnerParameterType.OFFSET:
+                offset = inner_par
+
+        scaling_factor_value = (
+            inner_parameter_values[scaling_factor.inner_parameter_id]
+            if scaling_factor is not None
+            else 1
+        )
+        offset_value = (
+            inner_parameter_values[offset.inner_parameter_id]
+            if offset is not None
+            else 0
+        )
+
+        # Get the data mask for the current observable.
+        observable_data_mask = (
+            scaling_factor.ixs if scaling_factor is not None else offset.ixs
+        )
```

Comment:
Is the logic here that the scaling indices are a subset of the offset indices? Or are they equal or `None`, and you just need it from one of them?

Reply:
I just need the data mask from one of them. Both of them have it. But also both of them could not exist (in the case of having only scaling or only offset in an observable)

Reply:
Can rewrite this as `scaling_factor.ixs or offset.ixs` then, should do the same thing, but fine as is

Reply:
Ohh true.
I forget about the `or` operator too often